example/src/MutableReference.hs

```diff
@@ -121,12 +120,12 @@ gens =
   -- , (5, return . Untyped $ Copy STuple0)
   ]
 
-returns :: MemStep resp refs -> SResponse () resp
-returns New         = SReference STuple0
-returns (Read  _)   = SResponse
-returns (Write _ _) = SResponse
-returns (Inc   _)   = SResponse
-returns (Copy _)    = SReference STuple0
+instance HasResponse MemStep where
+  response New         = SReference STuple0
+  response (Read  _)   = SResponse
+  response (Write _ _) = SResponse
+  response (Inc   _)   = SResponse
+  response (Copy _)    = SReference STuple0
```

Comment:
This could be:
```
instance HasResponse MemStep where
  response New   {} = SReference STuple0
  response Read  {} = SResponse
  response Write {} = SResponse
  response Inc   {} = SResponse
  response Copy  {} = SReference STuple0
```